cmd/clusterctl/validation/validate_pods.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type validationError struct {
+	name    string
+	message string
+}
+
+func ValidatePods(w io.Writer, c client.Client, namespace string) error {
+	fmt.Fprintf(w, "Validating pods in namespace %q\n", namespace)
+
+	pods, err := getPods(c, namespace)
+	if err != nil {
+		return err
+	}
+	return validatePods(w, pods, namespace)
+}
+
+func getPods(c client.Client, namespace string) (*corev1.PodList, error) {
+	pods := &corev1.PodList{}
+	if err := c.List(context.TODO(), client.InNamespace(namespace), pods); err != nil {
+		return nil, fmt.Errorf("Failed to get pods in namespace %q: %v", namespace, err)
```

Comment:
nit: error strings should start with a lower case letter. 

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Fixes:
https://github.com/kubernetes-sigs/cluster-api/pull/679/files#diff-53767fe464d92e82c3034f3769d689dfR47

https://github.com/kubernetes-sigs/cluster-api/pull/679/files#diff-53767fe464d92e82c3034f3769d689dfR56

https://github.com/kubernetes-sigs/cluster-api/pull/679/files#diff-53767fe464d92e82c3034f3769d689dfR94